src/main/java/frc/robot/Robot.java

```diff
@@ -124,32 +135,101 @@ public void robotInit() {
         swerve.runVelocityFieldRelative(
             () ->
                 new ChassisSpeeds(
-                    -controller.getLeftY() * SwerveSubsystem.MAX_LINEAR_SPEED,
-                    -controller.getLeftX() * SwerveSubsystem.MAX_LINEAR_SPEED,
-                    -controller.getRightX() * SwerveSubsystem.MAX_ANGULAR_SPEED)));
+                    -teleopAxisAdjustment(controller.getLeftY()) * SwerveSubsystem.MAX_LINEAR_SPEED,
+                    -teleopAxisAdjustment(controller.getLeftX()) * SwerveSubsystem.MAX_LINEAR_SPEED,
+                    -teleopAxisAdjustment(controller.getRightX())
+                        * SwerveSubsystem.MAX_ANGULAR_SPEED)));
     elevator.setDefaultCommand(elevator.setExtensionCmd(() -> 0.0));
     feeder.setDefaultCommand(feeder.runVoltageCmd(0.0));
+    carriage.setDefaultCommand(carriage.runVoltageCmd(0.0));
     intake.setDefaultCommand(intake.runVoltageCmd(10.0));
-    shooter.setDefaultCommand(shooter.runStateCmd(Rotation2d.fromDegrees(0.0), 0.0, 0.0));
+    shooter.setDefaultCommand(
+        shooter.runStateCmd(
+            () -> Rotation2d.fromDegrees(0.0), () -> flywheelIdleSpeed, () -> flywheelIdleSpeed));
     reactionBarRelease.setDefaultCommand(
         reactionBarRelease.setRotationCmd(Rotation2d.fromDegrees(0.0)));
 
-    // Controller bindings here
+    controller.setDefaultCommand(controller.rumbleCmd(0.0, 0.0));
+    operator.setDefaultCommand(operator.rumbleCmd(0.0, 0.0));
+
+    // Robot state management bindings
+    new Trigger(() -> carriage.getBeambreak() || feeder.getFirstBeambreak())
+        .debounce(0.25)
+        .whileTrue(
+            Commands.parallel(
+                intake.runVoltageCmd(0.0).withInterruptBehavior(InterruptionBehavior.kCancelSelf),
+                controller.rumbleCmd(1.0, 1.0).withTimeout(0.25)));
+    new Trigger(() -> currentTarget == Target.SPEAKER)
+        .whileTrue(Commands.parallel(carriage.runVoltageCmd(5.0), feeder.indexCmd()));
+    new Trigger(() -> currentTarget == Target.AMP)
+        .whileTrue(
+            Commands.either(
+                Commands.parallel(carriage.indexBackwardsCmd(), feeder.runVoltageCmd(-5.0)),
+                Commands.parallel(carriage.indexForwardsCmd(), feeder.runVoltageCmd(0.0)),
+                () -> feeder.getFirstBeambreak()));
+
+    // ---- Controller bindings here ----
+    controller.leftTrigger().whileTrue(intake.runVoltageCmd(10.0));
     controller
         .rightTrigger()
-        .whileTrue(shooter.runStateCmd(Rotation2d.fromDegrees(45.0), 50.0, 40.0));
-    controller.start().onTrue(Commands.runOnce(() -> swerve.setYaw(Rotation2d.fromDegrees(0))));
-
-    // Test binding for autoaim
+        .and(() -> currentTarget == Target.SPEAKER)
```

Comment:
maybe contrary to the formatting, i might put these on the same line, so it reads better like:
```suggestion
.rightTrigger().and(() -> currentTarget == Target.SPEAKER)
```